lib/matplotlib/backend_tools.py

```diff
@@ -111,6 +117,23 @@ def destroy(self):
         """
         pass
 
+    def state_changed_event(self, sender, canvasevent, data):
```

Comment:
Hmmm, another little problem, if we do it like this, you listen to the tools for events, but use navigation to fire them... just odd